src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -173,19 +173,32 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
         workers.clear();
         if (compound.hasKey(TAG_WORKER))
         {
-            final NBTTagList workersTagList = compound.getTagList(TAG_WORKER, Constants.NBT.TAG_COMPOUND);
-            for (int i = 0; i < workersTagList.tagCount(); ++i)
+            try
             {
-                final CitizenData data = getColony().getCitizen(workersTagList.getCompoundTagAt(i).getInteger(TAG_ID));
-                if (data != null)
+                final NBTTagList workersTagList = compound.getTagList(TAG_WORKER, Constants.NBT.TAG_COMPOUND);
+                for (int i = 0; i < workersTagList.tagCount(); ++i)
                 {
-                    data.setWorkBuilding(this);
-                    workers.add(data);
+                    final CitizenData data = getColony().getCitizen(workersTagList.getCompoundTagAt(i).getInteger(TAG_ID));
+                    if (data != null)
+                    {
+                        data.setWorkBuilding(this);
+                        workers.add(data);
+                    }
+                }
+            }
+            catch(final Exception e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)